store/store.go

```diff
@@ -720,15 +720,21 @@ func (s *Store) GetTreeStoreIDs() ([]string, error) {
 
 // GetRemote tries to retrieve a remote with the given ACIURL. found will be
 // false if remote doesn't exist.
-func (s *Store) GetRemote(aciURL string) (*Remote, bool, error) {
-	var remote *Remote
-	found := false
-	err := s.db.Do(func(tx *sql.Tx) error {
-		var err error
+func (s *Store) GetRemote(aciURL string) (remote *Remote, found bool, err error) {
+	err = s.db.Do(func(tx *sql.Tx) (err error) {
 		remote, found, err = GetRemote(tx, aciURL)
-		return err
+		if err != nil {
+			return
+		}
+
+		if !found {
+			return fmt.Errorf("couldn't find the remote with URL %q", aciURL)
```

Comment:
I think you can get rid of the multiple early returns and just assign `err` here. Not sure if it is idiomatic in go, but I find something like this easier to read:
```
if err == nil && !found {
  err = fmt.Errorf("couldn't find the remote with URL %q", aciURL)
}
return
```